File: exercises/week05/src/AbstractFactory/Scala/ParserFactoryProducer.scala
package AbstractFactory.Scala

object ParserFactoryProducer {
  def getFactory(factoryType: String): AbstractParserFactory = {
    factoryType match {
      case "NYCFactory" =>
        new NYCParserFactory
      case "LondonFactory" =>
        new LondonParserFactory
    }
  }
}

File: exercises/week05/src/AbstractFactory/Java/ParserFactoryProducer.java
package AbstractFactory.Java;

public class ParserFactoryProducer {

    public static AbstractParserFactory getFactory(String factory){

      AbstractParserFactory newFactory = null;
        switch(factory){
            case "NYC":     factory.equals("NYCFactory");
                            newFactory = new NYCParserFactory();
                            break;
            case "London":  factory.equals("LondonFactory");
                            newFactory = new LondonParserFactory();
                            break;
        }
        return newFactory;
    }
}
